sapi/fpm/fpm/fpm_sockets.c

```diff
@@ -396,9 +396,20 @@ static int fpm_socket_af_inet_listening_socket(struct fpm_worker_pool_s *wp) /*
 static int fpm_socket_af_unix_listening_socket(struct fpm_worker_pool_s *wp) /* {{{ */
 {
 	struct sockaddr_un sa_un;
+    size_t socket_length = sizeof(sa_un.sun_path);
+    size_t address_length = strlen(wp->config->listen_address);
 
 	memset(&sa_un, 0, sizeof(sa_un));
-	strlcpy(sa_un.sun_path, wp->config->listen_address, sizeof(sa_un.sun_path));
+	strlcpy(sa_un.sun_path, wp->config->listen_address, socket_length);
+
+    if (address_length >= socket_length) {
+        zlog(
+            ZLOG_WARNING,
+            "[pool %s] cannot bind to UNIX socket '%s' as path is too long (found length: %lu, maximal length: %lu), trying cut socket path instead '%s'",
```

Comment:
It would be good to try to add a test for this. It should be a bit generic as I think it might differ on FreeBSD so not rely on exact size but use some RegExp.

Reply:
`\[pool .+\] cannot bind to UNIX socket '.+' as path is too long \(found length: \d+, maximal length: \d+\), trying cut socket path instead '.+'`
some pattern like this?